NAS2D/Renderer/RendererOpenGL.cpp

```diff
@@ -81,6 +81,16 @@ namespace
 		const auto apiResult = glGetString(name);
 		return apiResult ? reinterpret_cast<const char*>(apiResult) : "";
 	}
+
+	void dumpGraphicsInfo()
+	{
+		// Spit out system graphics information.
+		std::cout << "\t- OpenGL System Info -" << std::endl;
+		std::cout << "\tVendor: " << glString(GL_VENDOR) << std::endl;
+		std::cout << "\tRenderer: " << glString(GL_RENDERER) << std::endl;
+		std::cout << "\tDriver Version: " << glString(GL_VERSION) << std::endl;
+		std::cout << "\tGLSL Version: " << glString(GL_SHADING_LANGUAGE_VERSION) << std::endl;
+	}
```

Comment:
Ultimately I'd like to get rid of this code from the NAS2D library. Though if dependent projects want to output this information, perhaps we should have a way of returning the information in a struct, and leave it to the calling project to decide how to output the data.

Reply:
It's up to you, but for my engine I don't give my downstream projects a choice. As part of logging system requirements, the first thing app creation does after loading the subsystems is to log the system description, including adapters:

```none
Initializing Logger: Data\Logs\game.log...
-----------------------------------------------------------
Operating System:                     Windows 64-bit
Processor Type:                                  x64
Socket Count:                                      1
Logical Processor Count:                          12
Installed RAM:                                  64.0 GB
Available RAM:                                  63.9 GB
-----------------------------------------------------------
ADAPTERS
-----------------------------------------------------------
Name:                        NVIDIA GeForce RTX 2060
Vendor ID:                                      10DE
Device ID:                                      1F08
Subsystem ID:                               37D11458
Revision:                                         A1
Video Memory:                                    5.8 GB
System Memory:                                   0.0 GB
Shared System Memory:                           32.0 GB
Adapter Type:                               Hardware                             
-----------------------------------------------------------
Name:                  Microsoft Basic Render Driver
Vendor ID:                                      1414
Device ID:                                        8C
Subsystem ID:                                      0
Revision:                                          0
Video Memory:                                    0.0 GB
System Memory:                                   0.0 GB
Shared System Memory:                           32.0 GB
Adapter Type:                               Software                             
-----------------------------------------------------------
Selected Adapter: NVIDIA GeForce RTX 2060
-----------------------------------------------------------
Available Display Dimensions:
1920x1080x60
1920x1080x59
1920x1080x50
1680x1050x60
1680x1050x59
1680x1050x50
1600x1200x60
1600x1024x60
1600x1024x59
1600x1024x50
1600x900x60
1366x768x60
1360x768x60
1280x1024x75
1280x1024x60
1280x960x75
1280x960x60
1280x800x75
1280x800x60
1280x768x75
1280x768x60
1280x720x60
1280x720x59
1280x720x50
1176x664x60
1176x664x59
1176x664x50
1152x864x75
1024x768x75
1024x768x60
800x600x75
800x600x60
720x576x50
720x480x60
720x480x59
640x480x75
640x480x60
640x480x59
-----------------------------------------------------------
1 Xbox controllers detected!
...Snip...
```